src/sorting/radix_sort.rs

```diff
@@ -0,0 +1,52 @@
+pub fn radix_sort(data: &mut [isize]) {
+    let mut buckets = vec![vec![]; 256];
+
+    for n in 0..4 {
+        for r in data as &[isize] {
+            let val = *r;
+            let mut which = (val >> (8 * n)) & 0xFFisize;
```

Comment:
this part is a bit hard to understand, we can use some comment here or try an easier version :)